Python/compile.c

```diff
@@ -2119,6 +2058,48 @@ wrap_in_stopiteration_handler(struct compiler *c)
     return SUCCESS;
 }
 
+static int
+compiler_type_params(struct compiler *c, asdl_typeparam_seq *typeparams)
+{
+    if (!typeparams) {
+        return SUCCESS;
+    }
+    Py_ssize_t n = asdl_seq_LEN(typeparams);
+
```

Comment:
The newest crash happens if `n==0`. Not guarded here.

Not sure if 0 should be allowed here, but either a check or an assert should be added.

Reply:
Thanks, I thought it would be NULL if there are no type params but apparently that's not true. I pushed a fix.